src/collective/cover/tests/test_textlinessortable_widget.py

```diff
@@ -158,20 +158,65 @@ def test_extract(self):
         self.assertDictEqual(extracted_value, expected)
 
     def test_utf8_custom_data(self):
```

Comment:
I was seeing the same error without customising the title, 
(in fact I found it difficult to reproduce when there WAS a custom title)
So for example, the following gives an error for me:

        title = u'El veloz murciélago hindú comía feliz cardillo y kiwi'
        obj = self.portal['my-image']
        obj.setTitle(title)
        obj.reindexObject()
        self.tile.populate_with_object(obj)

        self.assertIsInstance(self.tile.get_title(obj), unicode)
        self.assertEqual(self.tile.get_title(obj), title)

maybe this supplements what you've already done?